server.go

```diff
@@ -68,6 +70,8 @@ type Server struct {
 
 	OpenApiSpec openapi3.T // OpenAPI spec generated by the server
 
+	Listener net.Listener
```

Comment:
I don't see a need for the listener to be public, you added WithListener

Reply:
you are right now is private